File: ch6/my_cpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"

class MyCustomNode : public rclcpp::Node
{
public:
   MyCustomNode() : Node("my_node_name"), counter_(0)
   {
       timer_ = this->create_wall_timer(std::chrono::seconds(1), 
                                        std::bind(&MyCustomNode::print_hello, this));
   }

   void print_hello()
   {
       RCLCPP_INFO(this->get_logger(), "Hello %d", counter_);
       counter_++;
   }
private:
   int counter_;
   rclcpp::TimerBase::SharedPtr timer_;
};


int main(int argc, char **argv)
{
   rclcpp::init(argc, argv);
   auto node = std::make_shared<MyCustomNode>();
   rclcpp::spin(node);
   rclcpp::shutdown();
   return 0;
}

File: ch7/my_py_pkg/my_py_pkg/my_first_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

class MyCustomNode(Node):
   def __init__(self):
       super().__init__('my_node_name')
       self.counter_ = 0
       self.timer_ = self.create_timer(1.0, self.print_hello)

   def print_hello(self):
       self.get_logger().info("Hello " + str(self.counter_))
       self.counter_ += 1

def main(args=None):
   rclpy.init(args=args)
   node = MyCustomNode()
   rclpy.spin(node)
   rclpy.shutdown()

if __name__ == '__main__':
   main()
